### PR TITLE
Ready for universal CI

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

Fix all go vet / staticcheck errors

# Implementation

- updated go-ipld-prime to latest get code gen that does not have go vet errors
- mean we had to change one test due to a traversal order effect from now properly sorting map fields for DAG-JSON / DAG-CBOR
- re-codegen sections to resolve `go vet` Unkeyed fields errors
- resolve all static check issues
- remove one unneeded wrapper method

# For discussion

go-graphsync has some unreliable tests on Circle-CI, and I'm sure those will be accentuated when we switch over to Universal CI, as they were with go-bitswap. So we'll need to do seperate work there.

### CODE REVIEW

File: allocator/allocator_test.go [hunk 1]
```diff
@@ -382,8 +382,8 @@ type releasePeer struct {
 }
 
 type pendingResult struct {
-	p      peer.ID
-	amount uint64
+	p      peer.ID //nolint:structcheck
+	amount uint64  //nolint:structcheck
```
Comment (mvdan) [inline allocator/allocator_test.go:386]: what warning is this preventing? as far as I know our unified CI only enforces vet and staticcheck, so structcheck should not be a part of any of this.

**Review by mvdan — APPROVED** [review]
LGTM, except the nolint bit, which should be unnecessary and unrelated. If it isn't, please provide details, because I'm not sure I follow :)

File: allocator/allocator_test.go (hunk 1 above)
Comment (hannahhoward) [reply allocator/allocator_test.go:386]: it gets the CURRENT CI to pass :P

Comment (mvdan) [reply allocator/allocator_test.go:386]: :facepalm: need to rush to remove golangci-lint, then :P

Comment (hannahhoward) [reply allocator/allocator_test.go:386]: I'll remove the nolint comment in the PR where we switch over.

**Pull request merged**